### PR TITLE
Morton order helper functions

### PR DESCRIPTION
**Pull request by IanLilleyT**
Here's the start of Morton order helper functions. It only does 2D encode and decode right now.

Update: Now include 3D encode and decode as well

cc @ptrgags @lilleyse 

For context, these are some preliminary helper routines that will eventually be used in an implementation of the [`3DTILES_implicit_tiling`](https://github.com/CesiumGS/3d-tiles/pull/444) specification (still a WIP)

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @IanLilleyT!
* :heavy_check_mark: Signed CLA found.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

File: Source/Core/MortonOrder.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import Check from "./Check.js";
+import defined from "./defined.js";
+import DeveloperError from "./DeveloperError.js";
+
+/**
+ * Morton Order (aka Z-Order Curve) helper functions.
+ * @see {@link https://en.wikipedia.org/wiki/Z-order_curve}
+ * @see {@link http://graphics.stanford.edu/~seander/bithacks.html#InterleaveBMN}
```
Comment (ptrgags) [inline Source/Core/MortonOrder.js:8]: Could we move this reference about bit interleaving into the `removeSpacingBetweenBits`? feel like it's more helpful closer to the implementation

File: Source/Core/MortonOrder.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+ *
+ * @namespace MortonOrder
+ */
+var MortonOrder = {};
+
+/**
+ * @private
```
Comment (ptrgags) [inline Source/Core/MortonOrder.js:15]: I realize these are private methods, but would be nice to have a brief description of input and output.

File: Source/Core/MortonOrder.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+ * @param {Number} v
+ */
+function insertSpacingBetweenBits(v) {
```
Comment (ptrgags) [inline Source/Core/MortonOrder.js:18]: Can these functions be used in the future for interleaving 3D coordinates, or will they need to be modified?

**Review by ptrgags — COMMENTED** [review]
@IanLilleyT Looked over the code here, looks good, 

[Local Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=bZFRa4MwFIX/ysUnpRq3MsagVgbt3jb6UNhTXlK97eJiUpJo6Yb/fVG3kq6+JNybc879LmmZhpbjCTUsQeIJVmh4U5P3oRfSoBjqlZKWcYmaBtGCSipb53tT2iq50eVg/jV6zUFYKGmUQCLUwaW9uLrk8gBJkoxRe6Uh7NO4y7hbuCuDJ3fNZhF8UwlwEVSjoBoF1UUA4M/gMVQx0ACS3J2xD0mwn47z9SCKiFVbqx1MOI96EoCOyu6WeY2TzB7v44MPfEXjD+mxJqhK/KMaMLogDjJjzwLzcTuAZ14fnQEaLUJCUov1UTCLJt01xSdaUhgzLgCQpb41K3kLvFxOfCMUghnjXvaNEFv+hTTIs9Tpb6xCsX79TYtasHMv+7jPX8cmISRLXTnttEqJHdP/kn8A) to test [this table of values](https://en.wikipedia.org/wiki/Z-order_curve#Coordinate_values), works nicely!

didn't have much to say, just a couple minor documentation things and a question about how this will generalize to 3D

**Comment by IanLilleyT** [comment]
@ptrgags I addressed all of your feedback and added 3D Morton functions and more unit tests. Ready for another review.

**Review by ptrgags — APPROVED** [review]
Looks good, once CI passes I'll merge into the staging branch.

**Pull request merged**